### ISSUE TITLE
AOSP 10 GSI all variant releases

### ISSUE THREAD

**Issue by DarkestSpawn** [phhusson/treble_experimentations#976]
Unable to receive SMS all release variants. Initial setup caught up to sent messages while installing GSI and setting up Android but shortly after stops receiving. Toggling Airplane mode pulls incoming un-received texts for a few seconds than locks back up. Attempted United States and Automatic for Radio Band, and all options under Preferred Network Type. Issue persists even with RCS messaging setup on Google Messenger. Razer Phone 2 ATT network ATT phone APN default settings. Not sure how to narrow down correct info from LOGCAT

**Comment by phhusson** [comment]
Please provide `logcat -b all` (your logcat doesn't contain radio log)
and please have an SMS sent during logcat, and please write down the time at which the SMS has been sent.

**Comment by DarkestSpawn** [comment]
[Logcat.txt](https://github.com/phhusson/treble_experimentations/files/3965902/Logcat.txt)
21:51:55
Only copied the logcat info for 1 minute.

**Comment by DarkestSpawn** [comment]
[Logcat2.txt](https://github.com/phhusson/treble_experimentations/files/3965928/Logcat2.txt)
22:09:30
Rapid fire sending messages to localize error in logcat.

**Comment by phhusson** [comment]
Does SMS arrive if you force to 2g/3G?

**Comment by DarkestSpawn** [comment]
I have been unable to force 2g/3g. When I attempted rolling thru all Preferred Network types, It either disables the Radio, or only enables 4g.

**Comment by phhusson** [comment]
Try switching by calling in dialer *#*#4636#*#*

Le lun. 24 févr. 2020 à 01:13, DarkestSpawn <notifications@github.com> a
écrit :

**Comment by DarkestSpawn** [comment]
That's where I switched between preferred radio types (hidden testing --> phone info). Tried evdo, cdma, wcdma, umts, gsm, etc. I went thru all and was unable to force 2g or 3g.

**Comment by phhusson** [comment]
That's really not reassuring...

Well you can try enabling IMS
```
You'll need my latest release v212.
Download https://treble.phh.me/ims-q.apk
Install it just like you would install any apk, by either clicking on it from file manager, or with adb install
Download https://treble.phh.me/android.hardwa...ephony.ims.xml
Push it to /system/etc/permissions: adb root & adb remount & adb push android.hardware.telephony.ims.xml /system/etc/permissions/android.hardware.telephony.ims.xml
Reboot, unlock your device, wait one minute. Reboot again, unlock your device, wait one minute.
Reboot. Wait one minute, go in *#*#4636#*#* and see if IMS works.
```

**Comment by DarkestSpawn** [comment]
I did the above as stated. I swapped thru the preferred networks and still have issues. IMS status shows Registered. VoLTE: Available. VoWIFI: Unavailable.Video Calling: Unavailable. UT Interface: Available. Everytime I restart or disable then re-enable the radio the first sms works. After that it either fails to send or sends but never receives again.

**Comment by DarkestSpawn** [comment]
Well I guess I said that too soon. I rebooted for the 15th time, flipped to LTE only and my Texts flooded in from me. I enabled Chat Features as well. No idea why rebooting so many times worked. I attempted enabling the chat features somewhere around the 7th reboot. Thanks

**Comment by DarkestSpawn** [comment]
So I have been able recreate how to get texting to work. Reboots cause it to revert to not receive. BUT if I pull the sim card, reboot with it out, put it back in after it reboots, reboot once and sms sends and receives no issue.

**Comment by phhusson** [comment]
behaviour is same for VoLTE?

Le mar. 25 févr. 2020 à 07:46, DarkestSpawn <notifications@github.com> a
écrit :

**Comment by DarkestSpawn** [comment]
Correct. Can't make any calls until following the steps to get sms working.

**Comment by phhusson** [comment]
If you reboot, wait a minute, go in treble settings (no need to do anything
in it), then check IMS, is IMS okay? (So without plugging SIM out and in)

Le mar. 25 févr. 2020 à 08:59, DarkestSpawn <notifications@github.com> a
écrit :

**Comment by DarkestSpawn** [comment]
It shows IMS registered but I'm unable to receive messages or call out.

**Comment by phhusson** [comment]
Ok. Please give logcat when you remove/insert simcard

Le mar. 25 févr. 2020 à 09:17, DarkestSpawn <notifications@github.com> a
écrit :

**Comment by DarkestSpawn** [comment]
Regular logcat, or logcat -b all?

**Comment by phhusson** [comment]
Right -b all will probably be better

Le mar. 25 févr. 2020 à 09:36, DarkestSpawn <notifications@github.com> a
écrit :

**Comment by DarkestSpawn** [comment]
[logcat b-all.txt](https://github.com/phhusson/treble_experimentations/files/4248972/logcat.b-all.txt)
[logcat.txt](https://github.com/phhusson/treble_experimentations/files/4248974/logcat.txt)
I did both just in case

**Comment by DarkestSpawn** [comment]
Were they any help? Any pushes on 213 I should test? I see 213 added remote ADB from you.

**Comment by phhusson** [comment]
Could you check smsc in `*#*#4636#*#*`

**Comment by DarkestSpawn** [comment]
Wasn't able to get to this till today. Still having the same issues on 213. SMSC is blank. When I hit refresh it populates an 11 digit code in parentheses with a + at the beginning and after the parentheses I have a comma with 3 digits.

**Comment by DarkestSpawn** [comment]
Anything in 214 to test for the SMS issue?

**Comment by DarkestSpawn** [comment]
I have been running 214 for about a day now with the issue fix for SMS from issue [#1209](https://github.com/phhusson/treble_experimentations/issues/1209). I am able to receive and send both SMS/MMS.

**Comment by aykut19071993** [comment]
> I have been running 214 for about a day now with the issue fix for SMS from issue [#1209](https://github.com/phhusson/treble_experimentations/issues/1209). I am able to receive and send both SMS/MMS.

did you solve the problem?

**Comment by KhushrajRathod** [comment]
Try https://github.com/KhushrajRathod/VoLTE-Fix

**Comment by KhushrajRathod** [comment]
@DarkestSpawn @aykut19071993

**Comment by AL2009man** [comment]
> Try https://github.com/KhushrajRathod/VoLTE-Fix

Tried that, didn't work. but turning on 3G, I am able to receive SMS Messages. I guessing LTE is the issue here.

oh, and I'm running on AOSP 10 v220 Gapps with Moto G7 Play XT1952-6 using Spaceminer's TWRP as my Recovery Mode. 

(by the way, Fingerprint Sensor and Screen Lock doesn't work with Moto G7 Play.)

**Comment by phhusson** [comment]
(well open an issue for that...?)

**Comment by ferdinandwinters** [comment]
I also stop receiving SMS after a period of time. I can do the setup process but eventually, SMS stops working. I applied the "ims-q.apk" fix but that doesn't work entirely. Eventually, I lose mobile connection entirely with no SIM detected and must reboot my device. I also have no VOLTE but that's not really an issue.

The device is a Nokia 7 Plus with stock vendor.

**Comment by stephenobe** [comment]
I am having the exact same issue with a Razer Phone 2 on T-Mobile. I figured I would drop here first in case you didn't want a duplicate bug report, but I can make one if you would like. I have tried most of the steps above and wouldn't mind helping by running a logcat. I was using v. 221.

**Comment by lss4** [comment]
A little bump here as I'm having an interesting issue. I'm using Havoc 3.8 on my Razer Phone 2, with IMS installed systemlessly using a Magisk Module.

While VoLTE works, SMS do not work properly sometimes, that a message (such as SMS verification code) would take about 10 minutes to arrive, and if I'm to receive multiple messages, subsequent ones would arrive one by one every exactly 10 minutes since the first one.

I just noticed this yesterday. However, I don't remember the SMS always being this problematic, as sometimes I'm able to receive the message without any delay (or maybe it's because I didn't really notice that the message came later than usual, as these messages weren't required to be on time). I don't use/receive SMS very often, so this issue is not always noticeable to me.

I put my SIM card to a simpler device (which I don't think it's LTE) and I was able to get the necessary SMS received without issue and complete what I wanted to do. Not sure if my carrier is acting up, or it's just the usual issues with IMS on GSI...

**Comment by ferdinandwinters** [comment]
Try setting your mobile data to 3G instead of 4G and see if the SMS problem
is resolved.

**Comment by lss4** [comment]
> Try setting your mobile data to 3G instead of 4G and see if the SMS problem is resolved.

I tried. It did not work. I tried every possible options but none worked.

The only "option" that worked was to use a different device to receive the SMS verification codes and get the work done, then put it back.

So not sure if this is broken on the GSI side, or it's my phone's vendor. It's difficult to test as I don't receive SMS often, and that my plans don't include free SMS messages for me to conduct tests without incurring cost.

**Comment by JonnyTech** [comment]
Try without Magisk.

**Comment by danpawlik** [comment]
I got same issue - can not send or receive any SMS messages.
Installing new IMS package works: Treble Settings -> IMS features -> install IMS for Qualcomm -> install package.

**Comment by orionbg** [comment]
Hey everyone,
Sorry for reviving something old but I've been trying all available A13 and A14 GSI roms on my Xiaomi Mix Fold 2 and SMS receiving does not work! I can send without problems but not receiving any.
Tried all guides I could find, all IMS.apk variants I could get my hands on...
Any help would be great!
